### DISCUSSION TITLE
Any option to list only the key, not include the value to speed up the iteration?

### DISCUSSION THREAD

**Discussion by jliang000** [hse-project/hse#8] [Q&A] [answered]
Notice that memory usage is high during the cursor read to go through the kvs.

**Comment by smoyerx** [comment]
How are you measuring memory usage?  HSE uses the page cache for key data, and will use however much the OS will allow.  Since these pages are never dirty, the OS can simply reclaim them without writing as needed.

For iteration, are you using a cursor with a KVS storing multi-segment keys, and with a filter whose length is equal to or greater than the key prefix length?  Otherwise, cursor performance can be greatly reduced, and iteration will consume more page cache memory.

**Comment by jliang000** [comment]
Thanks for the answer. We use 'smem -twk' to monitor memory usage.  For the iteration, I use cursor read/next without a filter so I can list all the keys for each kvs. Is there any approach to just list the keys efficiently? we don't need cache read so far.

**Comment by smoyerx** [comment]
I don't think we have way to do this today.  @davidboles or @gsr-git can you confirm.

**Comment by gsramdasi** [comment, marked as answer]
That's correct @smoyergh . We currently do not have a way to iterate through just the keys without reading in the values.

**Comment by jliang000** [comment]
thanks for the confirmation.
